lib/internal/http2/core.js

```diff
@@ -3128,6 +3128,17 @@ function connect(authority, options, listener) {
 
   if (typeof listener === 'function')
     session.once('connect', listener);
+
+  debug('Http2Session connect', options.createConnection);
+  // Socket already has some buffered data - emulate receiving it
+  // https://github.com/nodejs/node/issues/35475
+  if (typeof options.createConnection === 'function') {
```

Comment:
I think what we want here is to check for pending data, like we do in `TLSSocket#initRead`:

```suggestion
  if (socket && socket.readableLength) {
```

Reply:
absolutely, thanks for noticing this at the last moment